src/clipping/weileratherton.jl

```diff
@@ -0,0 +1,450 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    WeilerAthertonClipping()
+
+The Wieler-Atherton algorithm for clipping polygons.
+
+## References
+
+* Weiler, K., & Atherton, P. 1977. [Hidden surface removal using polygon area sorting]
+  (https://dl.acm.org/doi/pdf/10.1145/563858.563896)
+"""
+struct WeilerAthertonClipping <: ClippingMethod end
+
+# VertexType distinguishes three different types of vertices used for constructing the structure
+# used by the Weiler-Atherton clipping algorithm. Normal type represents regular vertices, the
+# connecting points of edges of a rings. Entering and Exiting types represent intersections of
+# two ring edges, where one belongs to a clipping and other belongs to a clipped ring. The
+# Entering and Exiting types describe, from the perspective of the edges of the clipping ring,
+# whether it enters or exits the interior of the clipped ring.
+abstract type VertexType end
+abstract type Normal <: VertexType end
+abstract type Entering <: VertexType end
+abstract type Exiting <: VertexType end
+
+# Data structure for clipping the polygons. Fields left and right are used depending on the
+# VertexType, as designated with the helper functions. The data structure forms a directed
+# graph with each element always pointing to two elements.
+mutable struct RingVertex{VT<:VertexType,M<:Manifold,C<:CRS}
+  point::Point{M,C}
+  left::RingVertex{<:VertexType,M,C}
+  right::RingVertex{<:VertexType,M,C}
+
+  function RingVertex{VT,M,C}(point) where {VT<:VertexType,M<:Manifold,C<:CRS}
+    v = new(point)
+    v.left = v
+    v.right = v
+    v
+  end
+end
```

Comment:
```suggestion
mutable struct RingVertex{VT<:VertexType,P<:Point}
  point::P
  left::RingVertex{<:VertexType,P}
  right::RingVertex{<:VertexType,P}

  function RingVertex{VT,P}(point) where {VT<:VertexType,P<:Point}
    v = new(point)
    v.left = v
    v.right = v
    v
  end
end
```

@jnemanja could this be simplified further as above?